### PR TITLE
Enable funlen linter (part 2)

### PR DESCRIPTION
**Pull request by brojeg**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: [934](https://github.com/ydb-platform/ydb-go-sdk/issues/934)

## What is the new behavior?

-
-
-

### CODE REVIEW

**Comment by brojeg** [comment]
@rekby Please review.

File: internal/stack/record.go
Comment (asmyasnikov) [inline internal/stack/record.go:112]: Can you add a unit tests for newest functions?

File: retry/retry.go
Comment (rekby) [inline retry/retry.go:299]: no need extract simple error formatting

File: retry/retry.go
Comment (rekby) [inline retry/retry.go:314]: no need extract simple error formatting

**Review by rekby — CHANGES_REQUESTED** [review]
Hello, thanks for the work!
it is very good for your work for code style.

I see new part better, for example you add linter skip for one of functions, which really doesn't need to split.


I have  some for the before merge it.

**Comment by brojeg** [comment]
Is there any news on this PR?

**Comment by rekby** [comment]
> Is there any news on this PR?

Thanks for remind, I will plan to see the PR today.

File: internal/decimal/decimal.go
Comment (rekby) [inline internal/decimal/decimal.go:95]: New version has difference with old version of Parse in some cases.

please add fuzzing test for check equals of old and new Parse function results:

```
func FuzzParse(f *testing.F) {
	f.Fuzz(func(t *testing.T, s string, precision, scale uint32) {
		expectedRes, expectedErr := oldParse(s, precision, scale)
		res, err := Parse(s, precision, scale)
		if expectedErr == nil {
			require.Equal(t, expectedRes, res)
		} else {
			require.Error(t, err)
		}
	})
}
```

Where `oldParse` is copy of `Parse` from main branch.
Now result has differences at arguments "100", 0, 0.

You can past `oldParse` near the test in `decimal_test.go`, I will remove it from the branch after merge.

You can add better fuzz-test for better test long numbers instead of random strings only.

You can run fuzzing by command:
```
go test -fuzz=FuzzParse
```

https://go.dev/doc/security/fuzz/

**Review by rekby — CHANGES_REQUESTED** [review]
Good job, thanks.


See for one additional method for test equals of old and new function while refactoring.

File: internal/decimal/decimal.go
Comment (brojeg) [reply internal/decimal/decimal.go]: Used existing function parseSign

File: internal/decimal/decimal.go
Comment (brojeg) [reply internal/decimal/decimal.go:95]: added the provided test function FuzzParse and oldParse function to decimal_test.go

**Review by rekby — CHANGES_REQUESTED** [review]
Test FuzzParse is failed.
```
=== RUN   FuzzParse
--- FAIL: FuzzParse (27.30s)
=== RUN   FuzzParse/1279e0a916766c26
    decimal_test.go:366: 
        	Error Trace:	/Users/rekby/projects/yandex/ydb-go-sdk-workspace/ydb-go-sdk/internal/decimal/decimal_test.go:366
        	            				/Users/rekby/gosdk/go1.22.0/src/reflect/value.go:596
        	            				/Users/rekby/gosdk/go1.22.0/src/reflect/value.go:380
        	            				/Users/rekby/gosdk/go1.22.0/src/testing/fuzz.go:335
        	Error:      	Not equal: 
        	            	expected: 100000000000000000000000000000000000
        	            	actual  : 0
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,6 +2,3 @@
        	            	  neg: (bool) false,
        	            	- abs: (big.nat) (len=2) {
        	            	-  (big.Word) 3136633892082024448,
        	            	-  (big.Word) 5421010862427522
        	            	- }
        	            	+ abs: (big.nat) <nil>
        	            	 })
        	Test:       	FuzzParse/1279e0a916766c26
    --- FAIL: FuzzParse/1279e0a916766c26 (27.30s)

Expected :100000000000000000000000000000000000
Actual   :0
```

With same input as in my previous comment:
```
go test fuzz v1
string("100")
uint32(0)
uint32(0)
```

It mean that result of old and new functions have differences for string "100", zero scale and zero precision.

File: internal/decimal/decimal.go
Comment (rekby) [reply internal/decimal/decimal.go:114]: Now the function setSpecialValue do many things, instead only parse special values. 

Now it:
1. Detect negative values
2. Cut sign from string

And it has interface with difficult to understand: it change input "v" and return it.
Usually function return new value without change input, rarer change input without return it.

Change and return used for chaining calls within builder - for you can call Obj.Method1().Method2().Method3().... with one line.

please return separate function `parseSign` function for detect and cut off sign and reuse the function here and in parseNumber.

Or detect and cut sign before call `setSpecialValue` (better `parseSpecialValue`) and pass the sign to the function, then to `parseNumber` if you won't check the sign twice.

P.S. i think about check the sign twice better - it is very small check, but it more understandable to read, then many parameters.

**Review by rekby — CHANGES_REQUESTED** [review]
Hello, good work, thanks!
I don't found errors now.

It need only small cosmetic fix and resolve conflicts before merge

**Comment by brojeg** [comment]
I'm not quite sure how to resolve the particular merge conflict. 
The particular code that needs to be merged in extractLambdas function exists under parseFunctionName function in my PR.

**Comment by rekby** [comment]
> I'm not quite sure how to resolve the particular merge conflict. The particular code that needs to be merged in extractLambdas function exists under parseFunctionName function in my PR.

You can't merge it "automatically", because you have refactored the code.
You should see what mean changes in master branch, then re-implement the changes at own code. Then remove master's code block on merge.

**Comment by brojeg** [comment]
@rekby please help me to resolve the conflict on your side.
Due refactoring I've introduced the helper structure that do not exists in master. It's the only difference between master in my PR. 
Please accept my code.
<<<<<<< master
		details.structName = split[1]
=======
		structName = split[1]
>>>>>>> master

**Review by rekby — APPROVED** [review]
Thanks for great work!

File: internal/stack/record.go
Comment (rekby) [inline internal/stack/record.go:113]: Once more point:
record.go was 100% coveraged before refactor and 98.8% after.

One branch of the code not tested: point when `file` doesn't contain `/`.
Add test for this please and PR will be merge.

**Review by rekby — CHANGES_REQUESTED** [review]
Attention:

I have pushed commit to your branch for resolve merge conflict with master. Pull the branch before work with the code.

File: internal/stack/record.go
Comment (rekby) [reply internal/stack/record.go]: I fixed the issue by replace the code by function from standard library - for extrace file name from path.

**Review by rekby — APPROVED** [review]

**Pull request merged**
